public/app/features/variables/editor/reducer.ts

```diff
@@ -78,14 +78,10 @@ const variableEditorReducerSlice = createSlice({
       delete state.errors[action.payload.errorProp];
       state.isValid = Object.keys(state.errors).length === 0;
     },
-    changeVariableEditorExtended: (
-      state: VariableEditorState,
-      action: PayloadAction<{ propName: string; propValue: any }>
-    ) => {
-      // @ts-ignore - temp ignoring the errors now the state type is more strict
+    changeVariableEditorExtended: (state: VariableEditorState, action: PayloadAction<VariableEditorExtension>) => {
       state.extended = {
         ...state.extended,
-        [action.payload.propName]: action.payload.propValue,
+        ...action.payload,
```

Comment:
nit/future fear: Maybe this works now but in the future this can start to store some unknowns into state.

Reply:
Hmmm... I'm not sure if I follow?

I did originally have this as `state.extended = action.payload;`, but changed it to what it is now in the PR to more closely match the current behavior of extending what's already there.

I think spreading existing state in keeps behavior/semantics the same as before, but I also don't think there should be problems caused by just overriding it with the action payload.

Happy to defer to you on what the best thing to do here is.

Reply:
No worries